src/etc/cpu-usage-over-time.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# This is a small script that we use on CI to collect CPU usage statistics of
+# our builders. By seeing graphs of CPU usage over time we hope to correlate
+# that with possible improvements to Rust's own build system, ideally diagnosing
+# that either builders are always fully using their CPU resources or they're
+# idle for long stretches of time.
+#
+# This script is relatively simple, but it's platform specific. Each platform
+# (OSX/Windows/Linux) has a different way of calculating the current state of
+# CPU at a point in time. We then compare two captured states to determine the
+# percentage of time spent in one state versus another. The state capturing is
+# all platform-specific but the loop at the bottom is the cross platform part
+# that executes everywhere.
+#
+# # Viewing statistics
+#
+# All builders will upload their CPU statistics as CSV files to our S3 buckets.
+# These URLS look like:
+#
+#   https://$bucket.s3.amazonaws.com/rustc-builds/$commit/cpu-$builder.csv
+#
+# for example
+#
+#   https://rust-lang-ci2.s3.amazonaws.com/rustc-builds/68baada19cd5340f05f0db15a3e16d6671609bcc/cpu-x86_64-apple.csv
+#
+# Each CSV file has two columns. The first is the timestamp of the measurement
+# and the second column is the % of idle cpu time in that time slice. Ideally
```

Comment:
I presume this is the total idle % time; could we perhaps get per-thread/vCPU measurements? Fine to leave until later too, of course.

Reply:
The OSX/Linux implementations have this information, but for Windows we just have total times currently. It's possible to figure this out but afaik it's not all that useful in the sense that I can't think of a meaningful statistic to learn about individual cpus vs them all as an aggregate that we can act on

Reply:
My thought was a distinction between "one thread is doing work" and "all threads are busy, but none are all that active (e.g., network/IO heavy, presumably)". But yeah, I agree that at least initially not all that useful